llvm/include/llvm/CodeGen/RegAllocCommon.h

```diff
@@ -10,22 +10,25 @@
 #define LLVM_CODEGEN_REGALLOCCOMMON_H
 
 #include <functional>
+#include <llvm/CodeGen/Register.h>
 
 namespace llvm {
 
 class TargetRegisterClass;
 class TargetRegisterInfo;
+class MachineRegisterInfo;
 
 typedef std::function<bool(const TargetRegisterInfo &TRI,
-                           const TargetRegisterClass &RC)> RegClassFilterFunc;
+                           const MachineRegisterInfo &MRI, const Register Reg)>
+    RegClassFilterFunc;
```

Comment:
Rename this to RegAllocFilterFunc?

Reply:
It is already fixed in the original PR https://github.com/llvm/llvm-project/pull/93525. I will rebase it.